python-api/src/core/llm/interface.py

```
@@ -0,0 +1,40 @@
+    async def chat(self, request: ChatRequest) -> ChatResponse:
+        """Non-streaming chat completion."""
+
+    @abstractmethod
+    async def stream(self, request: ChatRequest) -> AsyncIterator[StreamChunk]:
+        """Streaming chat completion."""
+
+    async def list_models(self) -> list[dict[str, Any]]:
+        """Return available models. Override in providers that support it."""
+        return []
+
+    async def list_model_endpoints(self, model_id: str) -> list[dict[str, Any]]:
```

Ruff (0.15.2)

[warning] Unused method argument: `model_id`

(ARG002)